api/views.py

```diff
@@ -26,6 +26,20 @@ def post(self, request: request.Request):
 
         return response
 
+class UserAccountDeleteView(views.APIView):
+    permission_classes = (permissions.AllowAny,)
+    def delete(self, request):
+        try:
+            request_id = request.data["id"] 
+            user = User.objects.get(pk=request_id)
+            profile = Profile.objects.filter(user=user)
+            profile.delete()
+            user.delete()
```

Comment:
So we should get the requesting user from the request itself, not the data:
`user = request.user` 
This will eliminate the need for the try...catch
Also it should be possible to just delete the user, and let the profile delete through the cascade.

Reply:
so, you're saying the request is the user object itself, not the id of the user?

Reply:
Not quite, but request.user works just fine as the user

Reply:
It works fine you're right